core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -126,10 +126,10 @@ public ParsedQuery parse(QueryBuilder queryBuilder) {
             BytesReference bytes = queryBuilder.buildAsBytes();
             parser = XContentFactory.xContent(bytes).createParser(bytes);
             return parse(cache.get(), parser);
-        } catch (QueryParsingException e) {
+        } catch (QueryCreationException e) {
```

Comment:
I am conflicted on whether we should keep query parsing exception here. Anyways this method will go away right? Can we mark //norelease all the methods that we expect to be able to remove once we are done with the refactoring?